axelor-sale/src/main/java/com/axelor/apps/sale/service/loyalty/LoyaltyAccountServiceImpl.java

```diff
@@ -18,4 +43,21 @@ public Optional<LoyaltyAccount> getLoyaltyAccount(Partner partner, Company compa
     }
     return loyaltyAccount;
   }
+
+  protected Optional<LoyaltyAccount> getLoyaltyAccount(Partner partner, TradingName tradingName) {
+    Optional<LoyaltyAccount> loyaltyAccount = Optional.empty();
+    if (partner != null && tradingName != null) {
+      loyaltyAccount =
+          partner.getLoyaltyAccountList().stream()
+              .filter(account -> tradingName.equals(account.getTradingName()))
+              .findFirst();
+    }
+    return loyaltyAccount;
+  }
+
+  protected Optional<LoyaltyAccount> getLoyaltyAccount(Partner partner) {
+    return Optional.ofNullable(partner)
+        .map(Partner::getLoyaltyAccountList)
+        .map(account -> account.get(0));
```

Comment:
I think this method will not work if partner exists but has an empty loyalty account list.
Please check, but you should be able to replace this line by 

```suggestion
        .stream().findFirst();
```